src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotDisableCertificateValidationTests.cs

```diff
@@ -0,0 +1,448 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotDisableCertificateValidationTests : DiagnosticAnalyzerTestBase
+    {
+        [Fact]
+        public void TestLambdaDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Net;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        ServicePointManager.ServerCertificateValidationCallback += (a, b, c, d) => { return true; };
+    }
+}",
+            GetCSharpResultAt(8, 68, DoNotDisableCertificateValidation.Rule, "AnonymousFunction"));
```

Comment:
I am not sure we want to mention `AnonymousFunction` in the diagnostic message seen by end users. Tag @dotpaul